### ISSUE TITLE
CPU0 read msr failed

### ISSUE THREAD

**Issue by jamestang0219** [cocafe/msr-utility#10]
hi, im using the fantastic tool to fix my cpu issue. Following others experience, i write these instruction and execute in CMD:
`msr-cmd.exe -A setbit 0xc0010292 32 0
msr-cmd.exe -A rmwmask 0xc0010296 0x404040 0x0`

but the error occur:
`msr_rmw(): CPU0 read msr failed`

any suggestion? Thanks!

**Comment by cocafe** [comment]
Insufficient permission or the register is not supported by this CPU?

**Comment by cocafe** [comment]
> Insufficient permission or the register is not supported by this CPU?

or write-only register

**Comment by jamestang0219** [comment]
> > Insufficient permission or the register is not supported by this CPU?
> 
> or write-only register

thx for reply, my cpu is 7840hs, and i already use admin to execute cmd

**Comment by pexcn** [comment]
~~Same error. My CPU is Intel i7-8750H and i5-8250U.~~

~~Note:~~
~~1. I am using Administrator~~
~~2. Disabled `CFG Lock` in BIOS~~

~~So I think having enough permissions.~~

My bad...

**Comment by jamestang0219** [comment]
> ~Same error. My CPU is Intel i7-8750H and i5-8250U.~
> 
> ~Note:~ ~1. I am using Administrator~ ~2. Disabled `CFG Lock` in BIOS~
> 
> ~So I think having enough permissions.~
> 
> My bad...

how do u fix this issue?

**Comment by pexcn** [comment]
> how do u fix this issue?

@jamestang0219 Need to ensure that the address of the register is correct.
e.g.:
```bat
:: For i5-8250U
msr-cmd.exe -A read 0x606
```

**Comment by cocafe** [comment]
> > > Insufficient permission or the register is not supported by this CPU?
> > 
> > 
> > or write-only register
> 
> thx for reply, my cpu is 7840hs, and i already use admin to execute cmd

Try to read first, if it can't be read, this register may not support by your model, although this register can be RW on zen4 desktop.
[Here are some register references for zen4](https://github.com/cocafe/vcache-tray/blob/a6b3115894c6b9d47f53f6db7e188dbcf0538c1c/src/msr.c#L70)
You can wander around those repos control RYZEN to see whether some badass have decoded your platform. (keyword: ryzen, smu)
Also, grab a copy of latest AMD's PPR (Preliminary Processor Programming Reference) from somewhere to see whether these registers are documented publicly or not. (keyword: ppr, amd, ppr_RS_A0_nda_1.pdf(leaked, legacy maybe))

**Comment by cocafe** [comment]
> > > > Insufficient permission or the register is not supported by this CPU?
> > > 
> > > 
> > > or write-only register
> > 
> > 
> > thx for reply, my cpu is 7840hs, and i already use admin to execute cmd
> 
> Try to read first, if it can't be read, this register may not support by your model, although this register can be RW on zen4 desktop. [Here are some register references for zen4](https://github.com/cocafe/vcache-tray/blob/a6b3115894c6b9d47f53f6db7e188dbcf0538c1c/src/msr.c#L70) You can wander around those repos control RYZEN to see whether some badass have decoded your platform. (keyword: ryzen, smu) Also, grab a copy of latest AMD's PPR (Preliminary Processor Programming Reference) from somewhere to see whether these registers are documented publicly or not. (keyword: ppr, amd, ppr_RS_A0_nda_1.pdf(leaked, legacy maybe))

[One more, `winwring0` may be considered as malware on latest windows 11 version, although the error should be `failed to load driver`, you can try to shutdown that protection too.](https://github.com/cocafe/msr-utility/blob/6e2562bd66aff819f3216099fc1f5525cce7cb1a/readme.md?plain=1#L15)

**Comment by pexcn** [comment]
Hello @cocafe, Is there a way to implement `Sync MMIO` functionality similar to ThrottleStop?

![image](https://github.com/user-attachments/assets/3205dd4d-8ad6-4695-abb4-068ca4cf5256)

**Comment by cocafe** [comment]
> Hello @cocafe, Is there a way to implement `Sync MMIO` functionality similar to ThrottleStop?
> 
> ![image](https://private-user-images.githubusercontent.com/4590439/360359374-3205dd4d-8ad6-4695-abb4-068ca4cf5256.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzMTc1NDQsIm5iZiI6MTcyNDMxNzI0NCwicGF0aCI6Ii80NTkwNDM5LzM2MDM1OTM3NC0zMjA1ZGQ0ZC04YWQ2LTQ2OTUtYWJiNC0wNjhjYTRjZjUyNTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjJUMDkwMDQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjQ5ZjM3ZDRmZjczZjQwMmVlODYzNzI0NGE3OWU4NzVhZWFiY2ZhN2RmNTMyODBkNWY0MzExNDU2YmY0YzBkYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.oW4sZ8cNQlBgZPTI0cYD4ziEzK0QB5eruxmdw5m8N5E)

Yeah you can use my repo physmem, that MMIO is actually a memory address to write. Im outside now if u r interested i can shall my scripts for 12gen (mobile) later.

**Comment by pexcn** [comment]
> Yeah you can use my repo physmem, that MMIO is actually a memory address to write. Im outside now if u r interested i can shall my scripts for 12gen (mobile) later.

Thank you 😄

**Comment by cocafe** [comment]
> > Yeah you can use my repo physmem, that MMIO is actually a memory address to write. Im outside now if u r interested i can shall my scripts for 12gen (mobile) later.
> 
> Thank you 😄

Here U R https://github.com/cocafe/msr-scripts

**Comment by pexcn** [comment]
> Here U R https://github.com/cocafe/msr-scripts

OK, I will try it, thank you! :)
